numpy/core/src/common/get_attr_string.h

```diff
@@ -63,7 +71,7 @@ maybe_get_attr(PyObject *obj, char *name)
     if (tp->tp_getattr != NULL) {
         res = (*tp->tp_getattr)(obj, name);
         if (res == NULL) {
```

Comment:
nit: tempting to skip the new  function, and just add `&& PyErr_ExceptionMatches(PyExc_AttributeError)` to this `if` and the one below.

Reply:
I did the change, but the issue regarding `PyErr_ExceptionMatches` still remains.